src/plugins/proxy/src/infer_request.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/proxy/infer_request.hpp"
+
+#include "openvino/runtime/remote_context.hpp"
+#include "openvino/runtime/so_ptr.hpp"
+#include "remote_context.hpp"
+
+ov::proxy::InferRequest::InferRequest(ov::SoPtr<ov::IAsyncInferRequest>&& request,
+                                      const std::shared_ptr<const ov::ICompiledModel>& compiled_model)
+    : ov::IAsyncInferRequest(nullptr, nullptr, nullptr),
+      m_infer_request(std::move(request)),
+      m_compiled_model(compiled_model) {}
+void ov::proxy::InferRequest::start_async() {
+    m_infer_request->start_async();
+}
+
+void ov::proxy::InferRequest::wait() {
+    m_infer_request->wait();
+}
+
+bool ov::proxy::InferRequest::wait_for(const std::chrono::milliseconds& timeout) {
+    return m_infer_request->wait_for(timeout);
+}
+
+void ov::proxy::InferRequest::cancel() {
+    m_infer_request->cancel();
+}
+
+void ov::proxy::InferRequest::set_callback(std::function<void(std::exception_ptr)> callback) {
+    m_infer_request->set_callback(callback);
+}
+
+void ov::proxy::InferRequest::infer() {
+    m_infer_request->infer();
+}
+
+std::vector<ov::ProfilingInfo> ov::proxy::InferRequest::get_profiling_info() const {
+    return m_infer_request->get_profiling_info();
+}
+
+ov::Tensor ov::proxy::InferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
+    auto tensor = m_infer_request->get_tensor(port);
+    if (tensor.is<ov::RemoteTensor>()) {
+        auto remote_context = std::dynamic_pointer_cast<ov::proxy::RemoteContext>(m_compiled_model->get_context());
+        OPENVINO_ASSERT(remote_context);
+        tensor = remote_context->wrap_tensor(tensor.as<ov::RemoteTensor>());
+    }
+    return tensor;
+}
+
+void ov::proxy::InferRequest::set_tensor(const ov::Output<const ov::Node>& port, const ov::Tensor& tensor) {
+    m_infer_request->set_tensor(port, tensor);
+}
+
+std::vector<ov::Tensor> ov::proxy::InferRequest::get_tensors(const ov::Output<const ov::Node>& port) const {
+    auto tensors = m_infer_request->get_tensors(port);
```

Comment:
m_infer_request could be a gpu async infer request wrapper, call get_tensors here does not guarantee the so of gpu is wrapper in the tensor. so instead of changing m_infer_request to ov::SoPtr, it should be ov::Inferrequest structure.

Reply:
Ok agree, but it isn't a problem, because I have original so which was returned together with `ov::SoPtr::CompiledModel` and I can easily add it to the tensors. So thank you for the review, I will do it